### PR TITLE
[GSIP-185] web-resource extension

### PR DESCRIPTION
**Pull request by jodygarnett**
GeoServer Resource API provides programatic management of configuration files and icons. The REST API makes this functionality available for automation, we would like to make web-resource module an extension to offer this capability to web administration application.

See also:

* [GSIP-185](https://github.com/geoserver/geoserver/wiki/GSIP-185)
* [GEOS-9524](https://osgeo-org.atlassian.net/browse/GEOS-9524)

## Checklist

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira: [GEOS-9524](https://osgeo-org.atlassian.net/browse/GEOS-9524)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: doc/en/user/source/configuration/tools/resource/browser.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
+.. _web_resource:
+
+Resource Browser
+================
+
+The :guilabel:`Resource Browser` provides a tree showing configuration folders, along with actions to edit and mange resources.
+
+.. figure:: img/browser.png
+   
+   Catalog Bulk Load Tool
```
Comment (aaime) [inline doc/en/user/source/configuration/tools/resource/browser.rst]: Bulk load tool?

File: doc/en/user/source/extensions/controlflow/index.rst [hunk 2]
```diff
@@ -165,22 +165,6 @@ A complete example
...
-  # don't allow a single user to perform more than 6 requests in parallel
-  # (6 being the Firefox default concurrency level at the time of writing)
-  user=6
-  # don't allow the execution of more than 16 tile requests in parallel
-  # (assuming a server with 4 cores, GWC empirical tests show that throughput
-  # peaks up at 4 x number of cores. Adjust as appropriate to your system)
-  ows.gwc=16
-
-
+.. literalinclude:: controlflow.properties
```
Comment (aaime) [inline doc/en/user/source/extensions/controlflow/index.rst]: Why the unrelated changes?

File: doc/en/user/source/configuration/tools/resource/browser.rst (hunk 1 above)
Comment (jodygarnett) [reply doc/en/user/source/configuration/tools/resource/browser.rst]: it was undocumented

File: doc/en/user/source/extensions/controlflow/index.rst (hunk 2 above)
Comment (jodygarnett) [reply doc/en/user/source/extensions/controlflow/index.rst]: So I could reuse the code example

**Comment by aaime** [comment]
Ah one thing that came to mind, is the module accessible only to the full administrator? Or also to the workspace ones? If the latter, are the workspace admins locked down into their own workspace folders?

**Comment by jodygarnett** [comment]
> Ah one thing that came to mind, is the module accessible only to the full administrator? Or also to the workspace ones? If the latter, are the workspace admins locked down into their own workspace folders?

Just tested, only available to admin. Workspace admin does not see any tools.

image

**Comment by aaime** [comment]
NIce, then only the current (problematic, to me) position of the docs is left. If you don't agree let's maybe discuss on list, not sure everyone participates to PR feedbacks (I mean they should, but in practice there is no much activity).

**Comment by jodygarnett** [comment]
> NIce, then only the current (problematic, to me) position of the docs is left. If you don't agree let's maybe discuss on list, not sure everyone participates to PR feedbacks (I mean they should, but in practice there is no much activity).

I agree, this is just not the place to fix, would prefer a doc wide cleanup.
